generative_ai/rag/delete_corpus_example.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+PROJECT_ID = os.getenv("GOOGLE_CLOUD_PROJECT")
+
+
+def delete_corpus(corpus_name: str) -> None:
+    # [START generativeaionvertexai_rag_delete_corpus]
+
+    from vertexai.preview import rag
+    import vertexai
+
+    # TODO(developer): Update and un-comment below lines
+    # PROJECT_ID = "your-project-id"
+    # corpus_name = "projects/{PROJECT_ID}/locations/us-central1/ragCorpora/{rag_corpus_id}"
```

Comment:
Note: This can be either format:

``projects/{project}/locations/{location}/ragCorpora/{rag_corpus}``
        or ``{rag_corpus}``

Reply:
Yes, these files were written before as if you could use {rag_corpus}, but it’s not.

I've tried to use it without providing a full name. When we trying to use` rag.delete_corpus(name=corpus_name)`
It navigates to `rag_data.py` [here](https://github.com/googleapis/python-aiplatform/blob/86fc215cb854c006a0836a6e8f98e6d40c191fee/vertexai/preview/rag/rag_data.py#L189)

And then to [_gapic_utils.get_corpus_name(name)](https://github.com/googleapis/python-aiplatform/blob/86fc215cb854c006a0836a6e8f98e6d40c191fee/vertexai/preview/rag/utils/_gapic_utils.py#L349C2-L366C16)

Here it tries to match my corpus_id = "7991637538768945152" with regex pattern: 
`_VALID_RESOURCE_NAME_REGEX = "[a-z][a-zA-Z0-9._-]{0,127}"` witch presented [here](https://github.com/googleapis/python-aiplatform/blob/86fc215cb854c006a0836a6e8f98e6d40c191fee/vertexai/preview/rag/utils/_gapic_utils.py#L52)

The pattern does not match `7991637538768945152`, and it's clear that we cannot provide`{rag_corpus}` in its raw form. We need to use the full path instead.

 This is the reason why when I try to use `{rag_corpus}` as a parameter i faced with ValueError
```name must be of the format `projects/{project}/locations/{location}/ragCorpora/{rag_corpus}` or `{rag_corpus}`"```

**Once the library is updated, I will be able to fix my code.**